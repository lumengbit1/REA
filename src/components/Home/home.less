@borderstyle: 1px solid #000;
.homePage {
    display: flex;
    justify-content: space-between;
    height: 100vh;
    .resultsArea {
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
        width: 50%;
        margin-right: 10px;
        padding-top: 5%;
        border: @borderstyle;
    }
    .savedPropertiesArea {
        display: flex;
        flex: 1;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
        padding-top: 5%;
        border: @borderstyle;
    }
    .title {
        font-size: 30px;
        font-weight: bold;
        margin: 10px;
    }
    .block {
        border: 1px dashed #000;
        border-radius: 10px;
        padding: 20px;
        background-color: #fafbfa;
        min-width: 225px;
        min-height: 205px;
    }
}
