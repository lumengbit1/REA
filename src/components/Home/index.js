import React from 'react';
import CSSModules from 'react-css-modules';
import styles from './home.less';
import PropertyList from '../PropertyList';

function Home() {
    return (
        <div styleName="homePage">
            <div styleName="resultsArea">
                <div styleName="title">Results</div>
                <div styleName="block">
                    {/* <PropertyList area="results" /> */}
                    <PropertyList area="results" btnText="Add Property" btnClass="btnresults" />
                </div>
            </div>
            <div styleName="savedPropertiesArea">
                <div styleName="title">Saved Properties</div>
                <div styleName="block">
                    {/* <PropertyList area="saved" /> */}
                    <PropertyList area="saved" btnText="Remove Property" btnClass="btnsaved" />
                </div>
            </div>
        </div>
    );
}
export default CSSModules(Home, styles);
