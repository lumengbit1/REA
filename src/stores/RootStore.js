import { configure } from 'mobx';

import FormStore from './FormStore/FormStore';

configure({ enforceActions: 'always' });

class RootStore {
    constructor() {
        this.formStore = new FormStore(this);
    }
}

export default RootStore;
