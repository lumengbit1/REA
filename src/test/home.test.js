import React from 'react';
import RootStore from '../stores/RootStore';
import { Provider } from 'mobx-react';
import Home from '../components/Home';
import { render } from 'react-testing-library';

const rootStore = new RootStore();
describe('Render Test', function () {
    it('case: expect rendering correct', () => {
        const { container } = render(
            <Provider rootStore={rootStore}>
                <Home />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});
