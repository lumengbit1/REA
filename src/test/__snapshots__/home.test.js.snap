// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test case: expect rendering correct 1`] = `
<div>
  <div
    class="home-page"
  >
    <div
      class="results-area"
    >
      <div
        class="title"
      >
        'Results'
      </div>
      <div
        class="block"
      />
    </div>
    <div
      class="saved-properties-area"
    >
      <div
        class="title"
      >
        'Saved Properties'
      </div>
      <div
        class="block"
      />
    </div>
  </div>
</div>
`;
